src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1118,19 +1128,136 @@ map_injective f.injective u v
 
 end path
 
+/-! ## Converting between graphs -/
+
+namespace walk
+
+variables {G}
+
+/-- The walk `p` converted to lie in `H` -/
+@[simp] def induce : Π {u v : V} (p : G.walk u v) {H : simple_graph V}
+  (h : ∀ e, e ∈ p.edges → e ∈ H.edge_set), H.walk u v
+| _ _ (walk.nil) H h := walk.nil
+| _ _ (walk.cons a p) H h := by
+  { refine walk.cons _ (p.induce _);
+    simp only [walk.edges_cons, list.mem_cons_iff, forall_eq_or_imp, mem_edge_set] at h,
+    exact h.1, exact h.2, }
+
+variables {u v w : V} (p : G.walk u v) (q : G.walk v w)
+  {H : simple_graph V}
+  (hp : ∀ e, e ∈ p.edges → e ∈ H.edge_set)
+  (hq : ∀ e, e ∈ q.edges → e ∈ H.edge_set)
+
+lemma induce_id : p.induce (λ e ep, edges_subset_edge_set p ep) = p := by
+{ induction p,
+  simp only [induce],
+  simp only [p_ih, induce, eq_self_iff_true, heq_iff_eq, and_self], }
```

Comment:
Generally `begin`/`end` is used for multi-line proofs rather than `by ...`. Also, whenever you have multiple goals, we use `{` ... `}` to focus, which makes things easier to maintain since then you can tell where goals are supposed to be.
```suggestion
begin
  induction p,
  { simp only [induce] },
  { simp only [p_ih, induce, eq_self_iff_true, heq_iff_eq, and_self] },
end
```